### PR TITLE
tests: bluetooth: tester: add support for testing Multiple Notifications

### PR DESCRIPTION
**Pull request by KKopyscinski**
Enabled Multiple Notifications and added command for
bt_gatt_notify_multiple().

Increased possible number of connections to execute GATT/SR/GAN/BV-02-C

Signed-off-by: Krzysztof Kopyściński <krzysztof.kopyscinski@codecoup.pl>

### CODE REVIEW

File: tests/bluetooth/tester/src/gatt.c [hunk 1]
```diff
@@ -1908,6 +1909,67 @@ static void config_subscription(uint8_t *data, uint16_t len, uint16_t op)
...
+static void notify_mult(uint8_t *data, uint16_t len, uint16_t op)
+{
+	const struct gatt_cfg_notify_mult_cmd *cmd = (void *) data;
+	const size_t max_cnt = CONFIG_BT_L2CAP_TX_BUF_COUNT;
+	struct bt_gatt_notify_params params[max_cnt];
+	struct bt_conn *conn;
+	const size_t min_cnt = 1U;
+	int err = 0;
+	uint8_t status = BTP_STATUS_SUCCESS;
+	uint8_t attr_data_len = 0;
```
Comment (jori-nordic) [inline tests/bluetooth/tester/src/gatt.c:1928]: either `size_t` or `uint16_t` as that's what's used in `struct bt_gatt_notify_params`

File: tests/bluetooth/tester/src/gatt.c [hunk 2]
```diff
@@ -1908,6 +1909,67 @@ static void config_subscription(uint8_t *data, uint16_t len, uint16_t op)
...
+
+	for (uint16_t i = 0U; i < cmd->cnt; i++) {
+		attr_data_len = strtoul(server_db[cmd->attr_id[i] -
+				    server_db[0].handle].user_data, NULL, 16);
+		params[i].uuid = 0;
+		params[i].attr = &server_db[cmd->attr_id[i] -
+			server_db[0].handle];
+		params[i].data = &server_db[cmd->attr_id[i] -
+			server_db[0].handle].user_data;
+		params[i].len = sizeof(attr_data_len);
```
Comment (jori-nordic) [inline tests/bluetooth/tester/src/gatt.c:1954]: is this correct? I think you rather meant `params[i].len = attr_data_len;`

File: tests/bluetooth/tester/src/gatt.c [hunk 3]
```diff
@@ -1908,6 +1909,67 @@ static void config_subscription(uint8_t *data, uint16_t len, uint16_t op)
...
+
+		status = BTP_STATUS_FAILED;
+		goto rsp;
+	}
+
+	(void)memset(params, 0, sizeof(params));
+
+	for (uint16_t i = 0U; i < cmd->cnt; i++) {
+		attr_data_len = strtoul(server_db[cmd->attr_id[i] -
+				    server_db[0].handle].user_data, NULL, 16);
```
Comment (jori-nordic) [inline tests/bluetooth/tester/src/gatt.c:1948]: this doesn't read very well. I see you're making this calculation a bunch of times, could you just define it as local to the loop.

**Review by jori-nordic — CHANGES_REQUESTED** [review]

**Review by jori-nordic — DISMISSED** [review]

File: tests/bluetooth/tester/src/gatt.c [hunk 4]
```diff
@@ -1908,6 +1909,66 @@ static void config_subscription(uint8_t *data, uint16_t len, uint16_t op)
...
+		status = BTP_STATUS_FAILED;
+		goto rsp;
+	}
+
+	(void)memset(params, 0, sizeof(params));
+
+	for (uint16_t i = 0U; i < cmd->cnt; i++) {
+		struct bt_gatt_attr attr = server_db[cmd->attr_id[i] -
+			server_db[0].handle];
+		attr_data_len = strtoul(attr.user_data, NULL, 16);
```
Comment (hermabe) [inline tests/bluetooth/tester/src/gatt.c:1949]: Nit: Should have empty line after declaration.

File: tests/bluetooth/tester/src/bttester.h [hunk 5]
```diff
@@ -732,6 +732,13 @@ struct gatt_eatt_connect_cmd {
 
 #define GATT_READ_MULTIPLE_VAR		0x20
 
+#define GATT_NOTIFY_MULTIPLE		0x21
```
Comment (hermabe) [inline tests/bluetooth/tester/src/bttester.h:735]: The bit should be added to `supported_commands()` as well.

**Review by hermabe — CHANGES_REQUESTED** [review]
Looks good, just two minor comments.

**Review by hermabe — APPROVED** [review]

**Review by alwa-nordic — APPROVED** [review]

**Review by carlescufi — APPROVED** [review]

**Pull request merged**